lib/binaries/binary.ts

```diff
@@ -81,14 +81,20 @@ export abstract class Binary {
   /**
    * Gets the url to download the file set by the version. This will use the XML if available.
    * If not, it will download from an existing url.
+   *
+   * @param {string} version The version we are looking for. This could also be 'latest'.
+   * @param {opt_proxy} string Option to get proxy URL.
+   * @param {opt_ignoreSSL} boolean Option to ignore SSL.
    */
-  getUrl(version?: string): Promise<BinaryUrl> {
+  getUrl(version?: string, opt_proxy?: string, opt_ignoreSSL?: boolean): Promise<BinaryUrl> {
+    this.opt_proxy = opt_proxy || this.opt_proxy;
+    this.opt_ignoreSSL = opt_ignoreSSL || this.opt_ignoreSSL;
     if (this.alternativeDownloadUrl != null) {
       return Promise.resolve({url: '', version: ''});
     } else {
       return this.getVersionList().then(() => {
         version = version || Config.binaryVersions()[this.id()];
-        return this.configSource.getUrl(version).then(binaryUrl => {
+        return this.configSource.getUrl(version, opt_proxy, opt_ignoreSSL).then(binaryUrl => {
```

Comment:
They are used here https://github.com/angular/webdriver-manager/blob/83b7be0722c420e3440e66ba97b57b8d7311ea28/lib/binaries/config_source.ts#L41 for `XmlConfigSource` and https://github.com/angular/webdriver-manager/blob/83b7be0722c420e3440e66ba97b57b8d7311ea28/lib/binaries/config_source.ts#L141,L142 for `GithubApiConfigSource`.

But I am thinking if we could add these options in `config.json` for each CDN url, so that different config sources can have different options respectively.